src/components/toolbar/ToolBar.vue

```diff
@@ -1,99 +1,119 @@
-        :disabled="scale === initialScale"
-        data-test="zoom-reset"
-        v-b-tooltip.hover
-        :title="$t('Reset to initial scale')"
-      >
-        {{ $t('Reset') }}
-      </b-button>
-      <span class="btn btn-sm btn-secondary scale-value">{{ Math.round(scale*100) }}%</span>
-    </div>
+      <div class="btn-group btn-group-sm mr-2" role="group" aria-label="Zoom controls">
+        <b-button
+          class="btn btn-sm btn-secondary"
+          @click="scale += scaleStep"
+          data-test="zoom-in"
+          v-b-tooltip.hover
+          :title="$t('Zoom In')"
+        >
+          <font-awesome-icon :icon="plusIcon" />
+        </b-button>
+        <b-button
+          class="btn btn-sm btn-secondary"
+          @click="scale = Math.max(minimumScale, scale -= scaleStep)"
+          data-test="zoom-out"
+          v-b-tooltip.hover
+          :title="$t('Zoom Out')"
+        >
+          <font-awesome-icon :icon="minusIcon" />
+        </b-button>
+        <b-button
+          class="btn btn-sm btn-secondary"
+          @click="scale = initialScale"
+          :disabled="scale === initialScale"
+          data-test="zoom-reset"
+          v-b-tooltip.hover
+          :title="$t('Reset to initial scale')"
+        >
+          {{ $t('Reset') }}
+        </b-button>
+        <span class="btn btn-sm btn-secondary scale-value">{{ Math.round(scale*100) }}%</span>
+      </div>
 
-    <div class="btn-group btn-group-sm mr-2" role="group" aria-label="Additional controls">
-      <b-button
-        class="btn btn-sm btn-secondary ml-auto"
-        data-test="panels-btn"
-        @click="panelsCompressed = !panelsCompressed"
-        v-b-tooltip.hover
-        :title="panelsCompressed ? $t('Show Menus') : $t('Hide Menus')"
-      >
-        <font-awesome-icon :icon="panelsCompressed ? expandIcon : compressIcon" />
-      </b-button>
+      <div class="btn-group btn-group-sm mr-2" role="group" aria-label="Additional controls">
+        <b-button
+          class="btn btn-sm btn-secondary ml-auto"
+          data-test="panels-btn"
+          @click="panelsCompressed = !panelsCompressed"
+          v-b-tooltip.hover
+          :title="panelsCompressed ? $t('Show Menus') : $t('Hide Menus')"
+        >
+          <font-awesome-icon :icon="panelsCompressed ? expandIcon : compressIcon" />
+        </b-button>
+
+        <b-button
+          class="btn btn-sm btn-secondary mini-map-btn ml-auto"
+          data-test="mini-map-btn"
+          @click="miniMapOpen = !miniMapOpen"
+          v-b-tooltip.hover
+          :title="miniMapOpen ? $t('Hide Mini-Map') : $t('Show Mini-Map')"
+        >
+          <font-awesome-icon :icon="miniMapOpen ? minusIcon : mapIcon" />
+        </b-button>
+      </div>
 
       <b-button
+        v-if="breadcrumbData.length > 0"
         class="btn btn-sm btn-secondary mini-map-btn ml-auto"
         data-test="mini-map-btn"
-        @click="miniMapOpen = !miniMapOpen"
+        @click="saveBpmn"
```

Comment:
Same comment as in the modeler: why bind this to a (mostly empty) method where we can emit the event inline here in the handler?